api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -207,6 +197,23 @@ private void ensureConceptMapTypeIsSet(Concept concept) {
 		}
 	}
 
+	private void voidNameMakeSynonym(ConceptName conceptName) {
+		// Make the voided name a synonym, this would help to avoid
+		// having multiple fully specified or preferred
+		// names in a locale in case the name is unvoided
+		if (!conceptName.isSynonym()) {
+			conceptName.setConceptNameType(null);
+		}
+	}
+
+	private void falsifyLocalePreferred(ConceptName conceptName) {
```

Comment:
falsify does look so much tied to the method implementation details. I would go for something like makeLocaleNotPreferred or something else along those lines.